base/shared/src/main/scala/scalaz/data/IStream.scala

```diff
@@ -0,0 +1,22 @@
+package scalaz
+package data
+
+import scalaz.control.Inf
+
+
+trait IStreamModule {
+  type IStream[A]
+
+  def scons[A](a: =>A)(as: =>Id[IStream[A]]): IStream[A]
```

Comment:
If `IStream` already encapsulates laziness, then there should be no need to pass it by-name. Also, why wrapped in `Id`?

Reply:
True about the by-name, error on the `Id`. I've removed it.